pipeline/matching/evaluation/metrics.py

```diff
@@ -51,3 +53,49 @@ def count_selection_rate_per_generator(
             t = (selected, proposed)
             generator_selection_rate[key] = t
         return generator_selection_rate
+
+    def intent_distribution_in_selected_matches(
+        self, matches: List[Match]
+    ) -> Dict[str, int]:
+        total_count: Dict[str, int] = {}
+
+        for match in matches:
```

Comment:
Lines 64-67 and 70-73 are identical. While in matches should not contain items in match.metadata.rareIntents or match.metadata.intents if the match isn't using the intent-based match generators, to avoid duplicated effort it would be useful to check whether the value of 'match.metadata.generator` is within the accepted list of generators (`rareIntentsGenerator`, `similarIntentsGenerator`)